### DISCUSSION TITLE
Why does rbenv recommend using openssl@3 with ruby >= 3.1

### DISCUSSION THREAD

**Discussion by joshcooper** [rbenv/ruby-build#2067] [General]
I've tried building ruby 3.1.2 on macOS Catalina using `openssl@1.1` from homebrew using:

```
RUBY_CONFIGURE_OPTS="--disable-install-doc --with-openssl-dir=$(brew --prefix openssl@1.1)" rbenv install 3.1.2
```

but it always installs openssl 3.0.5 as defined in: https://github.com/rbenv/ruby-build/blob/7399e92f1fc9a10672c48ce5187a172d2cded2a3/share/ruby-build/3.1.2#L1

It's been recommended to use `openssl@3` instead of `openssl@1.1`, for example:

https://github.com/rbenv/ruby-build/discussions/1969#discussioncomment-3702729
https://github.com/rbenv/ruby-build/discussions/2053#discussioncomment-3729674
https://github.com/rbenv/ruby-build/discussions/2061#discussioncomment-3795920

However, my understanding is ruby >= 3.1 requires the [openssl 3.0 gem](https://github.com/ruby/openssl), but the native extension is compatible with both openssl 1.1.1 and 3.0 libraries. For example, ruby itself builds macOS against `openssl@1.1`:
https://github.com/ruby/ruby/blob/994b505ffb0bf9eb795525199b47697412a98abb/.github/workflows/macos.yml#L56

So I'm wondering why rbenv doesn't support Ruby 3.1 and openssl@1.1?

**Comment by mislav** [comment]
Hi, when you pass `--with-openssl-dir` via RUBY_CONFIGURE_OPTS explicitly like you did, ruby-build should _never_ download 
and compile any openssl version for you. https://github.com/rbenv/ruby-build/blob/55575fdb5ea862fda0545578520eec8f91264fc5/bin/ruby-build#L1135

Are you seeing that even though you pass that option, ruby-build downloads and compiles openssl 3.0.5 for you?

**Reply by joshcooper** [reply]
Thanks @mislav, it's probably user error on my part. I'll try to reproduce and post results or close this.

**Comment by joshcooper** [comment]
Not sure what the issue was, closing
